Feature (toml editor)

andrsj opened the pull request.

andrsj committed:
feature(toml_editor): Implement TOML editor

Implemented the TOML editor functionality
for applying new values on the fly
for future updating TOML files for our workflow
with sekaid config.toml variables

andrsj committed:
feature(toml_editor): Writing tests for `toml` package

Implemented test cases for cover all scenarious

MrLutik commented on pkg/ryokaicommon/utils/toml/toml.go:
```diff
@@ -0,0 +1,144 @@
+
+// searchNeededLine searches for the line index within the TOML configuration string
+// where the specified variable (within a tag, if provided) can be updated.
+func searchNeededLine(tag, name string, lines []string) (int, error) {
```
here i would rather go for `searchLine`

MrLutik commented on pkg/ryokaicommon/utils/toml/toml.go:
```diff
@@ -0,0 +1,144 @@
+}
+
+// isNumber checks if the given string is a numeric value.
+func isNumber(value string) bool {
```
This func is not toml specific. I would suggest to move it util or similar pkg

andrsj replied:
I will add if this method is used someone else

MrLutik commented on pkg/ryokaicommon/utils/toml/toml.go:
```diff
@@ -0,0 +1,144 @@
+	}
+}
+
+// isBool checks if the given string is a boolean value ("true" or "false").
```
This func is not toml specific. I would suggest to move it util or similar pkg

andrsj replied:
I will add if this method is used someone else

MrLutik commented on pkg/ryokaicommon/utils/toml/toml.go:
```diff
@@ -0,0 +1,144 @@
+		withinScope = tag == ""
+	)
+
+	for indexOfLine, line := range lines {
```
`for lineIndex` - sounds more right to me

MrLutik commented on pkg/ryokaicommon/utils/toml/toml.go:
```diff
@@ -0,0 +1,144 @@
+
+	lines := strings.Split(configStr, "\n")
+
+	searchedLineIndex, err := searchNeededLine(tag, name, lines)
```
`searchedLineIndex` -> `foundLineIndex`
as the function itself return int as rep of integer and err if fails. Logicaly it always find something.

MrLutik commented on pkg/ryokaicommon/utils/toml/toml.go:
```diff
@@ -0,0 +1,144 @@
+// if the variable could not be found or updated.
+func SetTomlVar(tomlValue Value, configStr string) (string, error) {
+	tag := strings.TrimSpace(tomlValue.Tag)
+	name := strings.TrimSpace(tomlValue.Name)
```
You added:
* `formatTag`
* `formatValue`
but not `formatName`. Is this on purpose? or we expect name to be immutable?

andrsj replied:
@MrLutik , name - is a name of property which has to be immutable because `sekaid` reads and expects for specific fields

andrsj committed:
feature(toml_editor): Fixing #5 PR issues

Fixing naming for several variables
Moving methods `IsBool` and `IsNumber` to `./utils/base` package
